airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -893,7 +893,10 @@ def generate_query(self) -> str:
             query += self.selected_fields
         else:
             query += "*"
-        query += f" from `{self.project_id}.{self.dataset_id}.{self.table_id}` limit {self.max_results}"
+        query += (
+            f" from `{self.project_id + '.' if self.project_id else ''}{self.dataset_id}"
```

Comment:
Does it work if we provide the hook project id explicitly?
```python
 query += f" from `{self.project_id or hook.project_id}.{self.dataset_id}.{self.table_id}` limit {self.max_results}"
```

Reply:
Yes, it should be equivalent. Soon I'll change it, as well as its unit test.